src/OpenTelemetry/Trace/IResolvingTracerProviderBuilder.cs

```diff
@@ -0,0 +1,26 @@
+// <copyright file="IResolvingTracerProviderBuilder.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace
+{
+    public interface IResolvingTracerProviderBuilder
```

Comment:
I don't like this because you are limiting yourself to these 2 methods for resolving services instead of just using the normal `IServiceProvider` with all the functionality that everyone is used to in this .NET world. `IServiceProvider` is in the base .NET code and has been forever. So you don't have to take any dependencies. I'd recommend that this interface just add a property for `IServiceProvider`.

Reply:
Thanks for the tip! Didn't realize (or had forgotten) `IServiceProvider` lives in System. I pushed an update so `IResolvingTracerProviderBuilder : IServiceProvider`. I didn't drop `IResolvingTracerProviderBuilder` completely because of the `IOptions<T>` case. In order to get options, you have to ask for IOptions or IOptionsMonitor or IOptionsSnapshot which is going to painful in exporters that might not have the dependency on options. So at the moment `IResolvingTracerProviderBuilder` exists to expose a helper to get those. Thoughts?

Reply:
Hmm... don't really like that the builder itself is implementing `IServiceProvider` instead of just exposing a property that is an `IServiceProvider`. Not really understanding the options thing. I'd like to experiment with the code, but haven't really gotten a chance to yet.